tests/utils/database.test.ts

```
@@ -0,0 +1,184 @@
+        const returnValue = await database.hasGameFile(
+          'src/utils/dataTest.json'
+        );
+        expect(returnValue).to.equal(true);
+      });
+    });
+    context('when file does not exist', () => {
+      it('expect hasGameFile to return false', async () => {
+        const returnValue = await database.hasGameFile(
+          'src/utils/dataTest.json'
+        );
+        expect(returnValue).to.equal(false);
+      });
+    });
+  });
+  context('load', () => {
+    context('when file exists and has data', () => {
+      before(async function () {
+        await writeFileSync(
+          'src/utils/data.json',
+          JSON.stringify({ foo: 'bar' })
+        );
+      });
+      after(async function () {
+        await unlinkSync('src/utils/data.json');
+      });
+      it('expect load to return data', async () => {
+        const returnValue = await database.load();
+        expect(returnValue).to.eql({ foo: 'bar' });
+      });
+    });
+    context('when file exists and has data - using file path over ride', () => {
+      before(async function () {
+        await writeFileSync(
+          'src/utils/dataTest.json',
+          JSON.stringify({ foo: 'bar' })
+        );
+      });
+      after(async function () {
+        await unlinkSync('src/utils/dataTest.json');
+      });
+      it('expect load to return data', async () => {
+        const returnValue = await database.load('src/utils/dataTest.json');
+        expect(returnValue).to.eql({ foo: 'bar' });
+      });
+    });
+    context('when file exists and has no data', () => {
+      before(async function () {
+        await writeFileSync('src/utils/data.json', '');
+      });
+      after(async function () {
+        await unlinkSync('src/utils/data.json');
+      });
+      it('expect load to return error', async () => {
+        const returnValue = await database.load();
+        expect(returnValue).to.have.own.property('error');
+      });
```

Should this case just throw instead of returning an error?

---

updated to return null